test/helper.js

```diff
@@ -68,22 +52,20 @@ helper.setupApi = function() {
     authServer = await base.testing.createMockAuthServer({
       port: 60407,
       clients:  defaultClients,
-      credentials: cfg.get('taskcluster:credentials')
+      credentials: cfg.taskcluster.credentials
     });
 
     // Create Hooks table
-    helper.Hook = data.Hook.setup({
-      account:      cfg.get('hooks:azureAccount'),
-      table:        cfg.get('hooks:hookTableName'),
-      credentials:  cfg.get('taskcluster:credentials'),
-      authBaseUrl:  cfg.get('taskcluster:authBaseUrl'),
-      signingKey:   cfg.get('hooks:tableSigningKey'),
-      cryptoKey:    cfg.get('hooks:tableCryptoKey'),
-      process:      'testing'
-    });
+    helper.Hook = await load('Hook', loadOptions);
+
+    // Remove all entities before each test
+    await helper.Hook.scan({}, {handler: hook => hook.remove()});
 
     helper.creator = new taskcreator.MockTaskCreator()
-    webServer = await bin.server(testProfile, {taskcreator: helper.creator});
+    webServer = await load('server', _.defaults({
+      Hook: helper.Hook,
+      taskcreator: helper.creator,
```

Comment:
Yeah, it wasn't so bad before... I just don't like special cases like:
```js
var creator = options.taskcreator || new taskcreator.TaskCreator({
        credentials:  cfg.get('taskcluster:credentials'),
  });
```
and 
```js
  // Start scheduler
  if (!options.noStart) {
    scheduler.start();
  }
```
Last one might not be fixed yet... but I think it's easily done by splitting it in two components `scheduler` (not started) and `schedule-hooks` (started, depends on `scheduler`), well something like that...